JASP-Engine/JASP/R/descriptives.R

```diff
@@ -329,7 +339,9 @@ Descriptives <- function(jaspResults, dataset, options) {
   if (base::is.factor(na.omitted) && (options$mean || options$mode || options$median || options$minimum || options$standardErrorMean || options$iqr || options$mad || options$madrobust || options$kurtosis || options$shapiro || options$skewness || options$percentileValuesQuartiles || options$variance || options$standardDeviation || options$percentileValuesPercentiles || options$sum || options$maximum)) {
     shouldAddNominalTextFootnote <- TRUE
   }
-    
+  
+  shouldAddIdenticalFootnote <- all(na.omitted[1] == na.omitted) && (options$skewness || options$kurtosis || options$shapiro)  
```

Comment:
so `==` is very strict... so strict that you'll notice it when `1 + 1e-16 == 1` is true but `1 + 2e-16 == 1` returns false. This basically means that this will never be true unless you actually duplicate the data. I'm not sure if this is the right implementation for this. I would consider at least the following two alternatives:

1. `isTRUE(all.equal(na.omitted[1], na.omitted))`. This uses `tolerance = sqrt(.Machine$double.eps)` which is also what most optimization algorithms use. Still I think this will rarely pass.
2. `var(na.omitted) < some_magic_number`. Most likely the failure is related to the absence of variance, so we could also just test that.

Both of these allow us to tweak the tolerance.

Reply:
> This basically means that this will never be true unless you actually duplicate the data.

Which is what we want, I think.

Basically I wanted to flag the condition for which `shapiro.test()` actually returns an error, and that is when `diff(range(x)) == 0`, which is effectively what `all(na.omitted[1] == na.omitted)` does except that the former would fail if x is a factor. If the range is very very small, the `shapiro.test()` just rescales the variable (which does not affect shapiro statistic). The alternatives which you proposed are nice but they would also flag cases for which we actually can compute the shapiro wilk, so I tend to think that my code is what we want for this case. I am happy to be convinced otherwise, but right now I don’t see a reason to have a more lenient check...